### PR TITLE
Issue #1501 print Runner.Run error messages

### PR DESCRIPTION
**Pull request by anjannath**

### CODE REVIEW

File: cmd/minishift/cmd/util/oc.go [hunk 1]
```diff
@@ -102,18 +103,19 @@ func SetOcContext(profileName string) error {
...
 	cmd := "config unset current-context"
+	errBuffer := new(bytes.Buffer)
 
 	ocRunner, err := oc.NewOcRunner(ocPath, filepath.Join(constants.Minipath, "machines", constants.MachineName+"_kubeconfig"))
 	if err != nil {
 		if glog.V(2) {
 			fmt.Println(fmt.Sprintf("%s", err.Error()))
 		}
-		return errors.New(fmt.Sprintf("Error unsetting current-context"))
+		return errors.New(fmt.Sprint("Error unsetting current-context"))
```
Comment (agajdosi) [inline cmd/minishift/cmd/util/oc.go]: Here the fmt.Sprint() is not necessary.

File: pkg/minishift/oc/oc.go [hunk 2]
```diff
@@ -74,9 +74,10 @@ func (oc *OcRunner) RunAsUser(command string, stdOut io.Writer, stdErr io.Writer
 // See also https://docs.openshift.org/latest/architecture/additional_concepts/authentication.html#authentication-impersonation
 func (oc *OcRunner) AddSudoerRoleForUser(user string) error {
 	cmd := fmt.Sprintf("adm policy add-cluster-role-to-user sudoer %s", user)
-	exitCode := oc.Run(cmd, nil, nil)
+	errorBuffer := new(bytes.Buffer)
+	exitCode := oc.Run(cmd, nil, errorBuffer)
 	if exitCode != 0 {
-		return errors.New("Unable to add sudoer role")
+		return fmt.Errorf("Unable to add sudoer role: %v", errorBuffer)
```
Comment (agajdosi) [inline pkg/minishift/oc/oc.go]: 👍

File: cmd/minishift/cmd/util/oc.go (hunk 1 above)
Comment (anjannath) [reply cmd/minishift/cmd/util/oc.go]: No, i thought it was but its not. Taking it out.

File: cmd/minishift/cmd/util/oc.go [hunk 3]
```diff
@@ -102,18 +103,19 @@ func SetOcContext(profileName string) error {
...
 		}
-		return errors.New(fmt.Sprintf("Error unsetting current-context"))
+		return errors.New("Error unsetting current-context")
 	}
 
-	exitCode := ocRunner.RunAsUser(cmd, nil, nil)
+	exitCode := ocRunner.RunAsUser(cmd, nil, errBuffer)
 	if exitCode != 0 {
-		return errors.New(fmt.Sprintf("Error during removing current context: %v", err.Error()))
+		return fmt.Errorf("Error during removing current context: %v", errBuffer)
```
Comment (praveenkumar) [inline cmd/minishift/cmd/util/oc.go]: 👍

File: cmd/minishift/cmd/util/oc.go [hunk 4]
```diff
@@ -102,18 +103,19 @@ func SetOcContext(profileName string) error {
...
 	cmd := "config unset current-context"
+	errBuffer := new(bytes.Buffer)
 
 	ocRunner, err := oc.NewOcRunner(ocPath, filepath.Join(constants.Minipath, "machines", constants.MachineName+"_kubeconfig"))
 	if err != nil {
 		if glog.V(2) {
 			fmt.Println(fmt.Sprintf("%s", err.Error()))
 		}
-		return errors.New(fmt.Sprintf("Error unsetting current-context"))
+		return errors.New("Error unsetting current-context")
```
Comment (praveenkumar) [inline cmd/minishift/cmd/util/oc.go]: 👍

**Comment by anjannath** [comment]
Rebased

**Comment by praveenkumar** [comment]
Since we already approved (me and @gbraad and @agajdosi ) I am going to merge it.

**Pull request merged**